specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
```

Comment:
Old specs clearly mentioned 1 to 8 octets. So while some implementations may allow it, some may not. It only matters for elements that have a default value but are not mandatory (rare). The `libebml` code allows size 0 for reading and writing.
In general the code was designed to not write the value if the value is the default value, regardless of the type. That means size 0 is always allowed for elements with a default value. That's from the original of saving space in the output file for stuff we don't need to store.
Since String elements were already allowed to be of 0-length, I'd say there shouldn't be much code that will break with non default 0-length elements. So I vote in favor of allowing it globally. And explaining what it means and when it can be used (default value otherwise not allowed).